ui/app/pages/send/send-content/send-gas-row/send-gas-row.component.js

```diff
@@ -79,7 +102,16 @@ export default class SendGasRow extends Component {
                 name: 'Changed Gas Button',
               },
             })
-            gasPriceButtonGroupProps.handleGasPriceSelection(...args)
+            return new Promise((resolve, reject) => {
```

Comment:
The function passed to the `Promise` constructor is executed synchronously and we're not doing anything async in the function so we can leave it out. We should be able to do something like:

```js
gasPriceButtonGroupProps.handleGasPriceSelection(...args)
if (maxModeOn) {
  this.setMaxAmount()
}
```

Reply:
For this one, I got rid of the promise and replaced with simple async/await, else it tries to set the max amount prematurely and causes an insufficient funds error.